File: src/dk/eightyplus/annoto/view/EditTextCustom.java
package dk.eightyplus.annoto.view;

import android.content.Context;
import android.util.AttributeSet;
import android.view.KeyEvent;
import android.widget.EditText;

/**
 * User: fries
 * Date: 20/03/14
 * Time: 22.44
 */
public class EditTextCustom extends EditText {

  private OnKeyListener keyListener;

  @SuppressWarnings("unused")
  public EditTextCustom(Context context) {
    super(context);
  }

  @SuppressWarnings("unused")
  public EditTextCustom(Context context, AttributeSet attrs) {
    super(context, attrs);
  }

  @SuppressWarnings("unused")
  public EditTextCustom(Context context, AttributeSet attrs, int defStyle) {
    super(context, attrs, defStyle);
  }

  public void setOnKeyBoardDownListener(OnKeyListener keyListener) {
    this.keyListener = keyListener;
  }

  @Override
  public boolean onKeyPreIme(int keyCode, KeyEvent event) {
    if (event.getKeyCode() == android.view.KeyEvent.KEYCODE_BACK) {
      if (keyListener != null && keyListener.onKey(this, keyCode, event)) {
        return true;
      }
    }
    return super.dispatchKeyEvent(event);
  }
}

File: app/src/main/java/dk/eightyplus/annoto/view/EditTextCustom.kt
package dk.eightyplus.annoto.view

import android.content.Context
import android.util.AttributeSet
import android.view.KeyEvent
import android.view.View
import android.widget.EditText

/**
 * User: fries
 * Date: 20/03/14
 * Time: 22.44
 */
class EditTextCustom : EditText {

        private var keyListener: View.OnKeyListener? = null

    constructor(context: Context) : super(context) {}

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {}

    constructor(context: Context, attrs: AttributeSet, defStyle: Int) : super(context, attrs, defStyle) {}

    fun setOnKeyBoardDownListener(keyListener: View.OnKeyListener) {
        this.keyListener = keyListener
    }

    override fun onKeyPreIme(keyCode: Int, event: KeyEvent): Boolean {
        if (event.keyCode == android.view.KeyEvent.KEYCODE_BACK) {
            if (keyListener?.onKey(this, keyCode, event) == true) {
                return true
            }
        }
        return super.dispatchKeyEvent(event)
    }
}
